### PR TITLE
"Manage access" modal: make sections collapsible

### PR DESCRIPTION
**Pull request by ajaypratap003**
https://issues.redhat.com/browse/MGDSTRM-6909

Design doc: https://marvelapp.com/prototype/hc64c9a/screen/84198391

image

image

image

image

### CODE REVIEW

**Comment by riccardo-forina** [comment]
Are sections staying open while interacting with the fields when adding new permissions? I remember that being a problem last time we tried adding this.

File: src/modules/Permissions/components/ManagePermissionsDialog/CreatePermissions.tsx [hunk 1]
```diff
@@ -56,6 +57,8 @@ export type CreatePermissionsProps = {
     | 'inline'
     | undefined;
   kafkaName: string | undefined;
+  isExpanded: boolean;
+  onChangeExpendedSection: (isExpanded: boolean) => void;
```
Comment (jgiardino) [inline src/modules/Permissions/components/ManagePermissionsDialog/CreatePermissions.tsx:61]: Really minor nit here, but can you fix the typo and change this to `onChangeExpandedSection` (here and throughout)?

File: src/modules/Permissions/dialogs/ManagePermissions/ManagePermissions.tsx [hunk 2]
```diff
@@ -116,6 +116,14 @@ export const ManagePermissionsModal: React.FC<
...
   const [isOpenPreCancelModal, setIsOpenPreCancelModal] =
     useState<boolean>(false);
+  const [
+    isExpandedExistingPermissionSection,
+    setIsExpandedExistingPermissionSection,
+  ] = useState<boolean>(false);
+  const [
+    isExpandedAssignPermissionsSection,
+    setIsExpandedAssignPermissionsSection,
+  ] = useState<boolean>(false);
```
Comment (jgiardino) [inline src/modules/Permissions/dialogs/ManagePermissions/ManagePermissions.tsx:126]: I'm not sure if these values represent the default state, but when these contents first load in modal, you should see:

- **Review existing permissions** is **collapsed**
- **Assign permissions** is **expanded**

**Comment by jgiardino** [comment]
@ajaypratap003 - I posted some comments, but I'm not sure how to move forward on this one. The epic that contains the parent link talks about refactoring the existing UI by adding it to storybook, but this is skipping that step in the process. I think part of the reason for wanting this in storybook before adding it here was so that we could test how it behaves in the storybook preview and make sure we've addressed all issue before pushing it to stage. 

I'm going to defer to @riccardo-forina on whether he wants to approve this for merging or not. For me, not having the preview available in storybook makes me less confident in pushing changes to production.

**Comment by riccardo-forina** [comment]
Agree on focusing the efforts on the ui-components repo

File: src/modules/Permissions/dialogs/ManagePermissions/ManagePermissions.tsx (hunk 2 above)
Comment (ajaypratap003) [reply src/modules/Permissions/dialogs/ManagePermissions/ManagePermissions.tsx:126]: @jgiardino  yes, make sense. I'll update default state for assign permission. Thanks for the feedback.

**Comment by ajaypratap003** [comment]
> @ajaypratap003 - I posted some comments, but I'm not sure how to move forward on this one. The epic that contains https://issues.redhat.com/browse/MGDSTRM-6909 talks about refactoring the existing UI by adding it to storybook, but this is skipping that step in the process. I think part of the reason for wanting this in storybook before adding it here was so that we could test how it behaves in the storybook preview and make sure we've addressed all issues before pushing it to stage.
> 
> I'm going to defer to @riccardo-forina on whether he wants to approve this for merging or not. For me, not having the preview available in storybook makes me less confident in pushing changes to production.

@jgiardino ExpandableSection is new feature and I don't think so there is any dependency on storybook. 
ExpandableSection can be integrate in storybook when `Manage Permissions Modal` will compete. ExpandableSection is working fine and we can merge if you are fine. 
I don't have any concern if you want to see ExpandableSection in storybook first and want to hold this change.

**Comment by ajaypratap003** [comment]
> Are sections staying open while interacting with the fields when adding new permissions? I remember that being a problem last time we tried adding this.

This is working fine as expected. It would be better if can try it on your machine.

**Comment by riccardo-forina** [comment]
Let's implement this on the ui-components repo.

**Pull request closed without merging**